frontend/src/metabase/models/containers/ModelDetailPage/ModelDetailPage.unit.spec.tsx

```diff
@@ -297,6 +379,155 @@ describe("ModelDetailPage", () => {
+          const model = getModel();
+          const action = createMockQueryAction({ model_id: model.id() });
+          await setupActions({ model, actions: [action] });
+
+          expect(screen.getByText(action.name)).toBeInTheDocument();
+          expect(screen.getByText(TEST_QUERY)).toBeInTheDocument();
+        });
+
+        it("lists existing implicit actions", async () => {
+          const model = getModel();
+          await setupActions({
+            model,
+            actions: createMockImplicitCUDActions(model.id()),
+          });
+
+          expect(screen.getByText("Create")).toBeInTheDocument();
+          expect(screen.getByText("Update")).toBeInTheDocument();
+          expect(screen.getByText("Delete")).toBeInTheDocument();
+        });
+
+        it("allows to create a new query action", async () => {
+          const model = getModel();
+          await setupActions({
+            model,
+            actions: [createMockQueryAction({ model_id: model.id() })],
+          });
+
+          userEvent.click(screen.getByRole("button", { name: "New action" }));
+
+          expect(screen.getByTestId("mock-action-editor")).toBeVisible();
+        });
+
+        it("allows to edit a query action", async () => {
+          const model = getModel();
+          const action = createMockQueryAction({ model_id: model.id() });
+          await setupActions({ model, actions: [action] });
+
+          const listItem = screen.getByRole("listitem", { name: action.name });
+          userEvent.click(within(listItem).getByLabelText("pencil icon"));
+
+          expect(screen.getByTestId("mock-action-editor")).toBeVisible();
+        });
+
+        it("allows to create implicit actions", async () => {
+          const createActionSpy = jest.spyOn(ActionsApi, "create");
+          const model = getModel();
+          const action = createMockQueryAction({ model_id: model.id() });
+          await setupActions({ model, actions: [action] });
+
+          userEvent.click(screen.getByTestId("new-action-menu"));
+          userEvent.click(screen.getByText("Create basic actions"));
+
+          await waitFor(() => {
+            expect(createActionSpy).toHaveBeenCalledWith({
+              name: "Create",
+              type: "implicit",
+              kind: "row/create",
+              model_id: model.id(),
+            });
+          });
+          expect(createActionSpy).toHaveBeenCalledWith({
+            name: "Update",
+            type: "implicit",
+            kind: "row/update",
+            model_id: model.id(),
+          });
+          expect(createActionSpy).toHaveBeenCalledWith({
+            name: "Delete",
+            type: "implicit",
+            kind: "row/delete",
+            model_id: model.id(),
+          });
+        });
+
+        it("allows to create implicit actions from the empty state", async () => {
+          const createActionSpy = jest.spyOn(ActionsApi, "create");
```

Comment:
Can we refactor this and assert on `nock`-mocked endpoint being called? You could also add an appropriate helper in `server-mocks` if it makes sense.